lib/pages/infoPages/vendorPage.dart

```diff
@@ -26,28 +38,63 @@ class _VendorPageWidgetState extends State<VendorPageWidget> {
   FirebaseUser user;
   DatabaseReference _userRef;
 
+  //Location variables
+  final _locationService = Geolocator();
+  Position currentLocation;
+
   //Declare contextual variables
   AppState appState;
   ThemeData theme;
 
   bool following = false;
 
+  Deals deals = Deals();
+  DatabaseReference _dealsRef;
+
   @override
   void initState() {
     super.initState();
+    currentLocation = widget.location;
     initialize();
   }
 
   void initialize() async {
     user = await _auth.currentUser();
     _userRef = FirebaseDatabase().reference().child("Users").child(user.uid);
+    _dealsRef = FirebaseDatabase().reference().child("Deals");
     _userRef.child("following").onValue.listen((data){
       if (data.snapshot != null){
         setState(() {
           following = data.snapshot.value[widget.vendor.key]?? false;
         });        
       }
     });
+    try {
+      var serviceStatus = await _locationService.checkGeolocationPermissionStatus();
+      if (serviceStatus == GeolocationStatus.granted) {//we should have permission but maybe they turned it off since they got here
+        _locationService.getPositionStream(LocationOptions(accuracy: LocationAccuracy.best)).listen((Position result) async {
+          if (this.mounted){
+            setState(() {
+              currentLocation = result;
+            });
+          }
+        });
+      }
```

Comment:
This can be moved out of the try block. Just the await service throws a PlatformException

Reply:
okay!

Reply:
Fixed in latest commit.